algorithms_task_6/student/include/impl.h

```diff
@@ -18,4 +18,9 @@ void insert_to_sorted(Sequence& target, const FwIt sourceBegin, const FwIt sourc
     static_assert(std::is_const<typename std::remove_reference<
                           typename std::iterator_traits<FwIt>::reference>::type>::value,
                       "FwIt must point to const data");
+    for(auto iter = sourceBegin; iter != sourceEnd; ++iter) {
+        auto it = std::upper_bound(target.begin(), target.end(), *iter);
+        target.insert(it, *iter);
+        //auto found = std::find_if(target.begin, target.end(), [&sourceBegin](Sequence& item){return item > static_cast<Sequence>(*sourceBegin);});
```

Comment:
find is a linear search, not effective
lower bound vs upper bound will make difference if out range contains equal values